JWT/JWT.cs

```diff
@@ -18,17 +20,31 @@ public enum JwtHashAlgorithm
     /// </summary>
     public static class JsonWebToken
     {
-        private static Dictionary<JwtHashAlgorithm, Func<byte[], byte[], byte[]>> HashAlgorithms;
+        private static readonly IDictionary<JwtHashAlgorithm, Func<byte[], string, byte[], byte[]>> HashAlgorithms;
         private static JavaScriptSerializer jsonSerializer = new JavaScriptSerializer();
 
         static JsonWebToken()
         {
-            HashAlgorithms = new Dictionary<JwtHashAlgorithm, Func<byte[], byte[], byte[]>>
+            HashAlgorithms = new Dictionary<JwtHashAlgorithm, Func<byte[], string, byte[], byte[]>>
+            {
+                { JwtHashAlgorithm.HS256, (key, _, value) => { using (var sha = new HMACSHA256(key)) { return sha.ComputeHash(value); } } },
+                { JwtHashAlgorithm.HS384, (key, _, value) => { using (var sha = new HMACSHA384(key)) { return sha.ComputeHash(value); } } },
+                { JwtHashAlgorithm.HS512, (key, _, value) => { using (var sha = new HMACSHA512(key)) { return sha.ComputeHash(value); } } },
+                { JwtHashAlgorithm.RS256, (key, secret, value) => RS256(key, secret, value) }
+            };
+        }
+
+        private static byte[] RS256(byte[] key, string secret, byte[] value)
+        {
+            var cert = new X509Certificate2(key, secret);
```

Comment:
So the "key" here is actually a byte array containing a certificate, I guess?  What format does it need to be in?  Is it a custom serialization specific to the X509Certificate class or is it some standard certificate container format or something?

Perhaps we should let folks pass in a X509Certificate2 (or just the private key?) rather than a key and a secret.  That way they can get the cert however they want (and they only have to pass in one thing instead of two).